tests/cstor/volume/provision_bulk_test.go

```diff
@@ -0,0 +1,263 @@
+		bulkCount     = 10
+		pvcObjList    *v1.PersistentVolumeClaimList
+		pvcObjTpl     *v1.PersistentVolumeClaim
+	)
+
+	When("creating resources required by test", func() {
+		It("should create a spc and storageclass", func() {
+
+			By("building spc")
+			spcObj = spc.NewBuilder().
+				WithGenerateName(spcName).
+				WithDiskType(string(apis.TypeSparseCPV)).
+				WithMaxPool(cstor.PoolCount).
+				WithOverProvisioning(false).
+				WithPoolType(string(apis.PoolTypeStripedCPV)).
+				Build().Object
+
+			By("creating above spc")
+			spcObj, err = ops.SPCClient.Create(spcObj)
+			Expect(err).To(BeNil(), "while creating spc {%s}", spcName)
+
+			By("verifying healthy csp count")
+			cspCount := ops.GetHealthyCSPCountEventually(spcObj.Name, cstor.PoolCount)
+			Expect(cspCount).To(Equal(true), "while checking csp health status")
+
+			By("building cas config with above spc")
+			CASConfig := strings.Replace(
+				openebsCASConfigValue, "$spcName", spcObj.Name, 1,
+			)
+			CASConfig = strings.Replace(
+				CASConfig, "$count", strconv.Itoa(cstor.ReplicaCount), 1,
+			)
+			annotations[string(apis.CASTypeKey)] = string(apis.CstorVolume)
+			annotations[string(apis.CASConfigKey)] = CASConfig
+
+			By("building a storageclass")
+			scObj, err = sc.NewBuilder().
+				WithGenerateName(scName).
+				WithAnnotations(annotations).
+				WithProvisioner(openebsProvisioner).Build()
+			Expect(err).ShouldNot(
+				HaveOccurred(),
+				"while building storageclass {%s}", scName,
+			)
+
+			By("creating above storageclass")
+			scObj, err = ops.SCClient.Create(scObj)
+			Expect(err).To(BeNil(), "while creating storageclass {%s}", scName)
+		})
+	})
+
+	// Actual tests begin here !!!
+	When("cstor pvcs are created", func() {
+		It("should create cstor volumes", func() {
+
+			By("building a pvc template")
+			pvcObjTpl, err = pvc.NewBuilder().
+				WithGenerateName(pvcNamePrefix).
+				WithNamespace(nsObj.Name).
+				WithStorageClass(scObj.Name).
+				WithAccessModes(accessModes).
+				WithCapacity(capacity).
+				WithLabels(
+					map[string]string{
+						"bulk.delete": nsObj.Name,
+					},
+				).
+				Build()
+			Expect(err).ShouldNot(
+				HaveOccurred(),
+				"while building pvc template {%s} in namespace {%s}",
+				pvcNamePrefix,
+				nsObj.Name,
+			)
+
+			By("building a list of pvcs")
+			pvcObjList, err = pvc.ListBuilderFromTemplate(pvcObjTpl).
+				WithCount(bulkCount).
+				APIList()
+
+			Expect(err).ShouldNot(
+				HaveOccurred(),
+				"while building a list of pvcs {%s} in namespace {%s}",
+				pvcNamePrefix,
+				nsObj.Name,
+			)
+
+			By("creating above pvcs in a bulk")
+			pvcObjList, err = ops.PVCClient.
+				WithNamespace(nsObj.Name).
+				CreateCollection(pvcObjList)
+			Expect(err).To(
+				BeNil(),
+				"while creating pvcs {%s} in namespace {%s}", pvcNamePrefix, nsObj.Name,
+			)
+
+			podCount := ops.GetPodRunningCountEventually(
+				openebsNamespace,
+				"openebs.io/storage-class="+scObj.Name,
```

Comment:
can we make use of the label that was set earlier instead of replying on `openebs.io/storage-class` label?
refer to my earlier comment

Reply:
No as the above label is for pvcs and here we are fetching the pods. Pods don't have the above label.